**PR title**: "community: Avoid cl100k_base encoding for Text embedding engines compatibility with LM Studio"

karkaratz opened the pull request:
- **Description:** With this update, users can set as False tiktoken_enabled when they ned it. By default, the value is set to True (so, not changing current behavior).
    - **Issue:** LM studio provide, by default, nomic-ai/nomic-embed-text-v1.5-GGUF for embedding. When calling the exposed API, with tiktoken_enabled=True, the client sends integers as embedding requests. The API expects Strings and it returns errors: [ERROR] 'input' field must be a string or an array of strings
    - **Dependencies:** 
    - **Twitter handle:** 


- [x] **Add tests and docs**: 
[test.txt](https://github.com/langchain-ai/langchain/files/15153763/test.txt)

Commit:
With this update, users can set as False tiktoken_enabled when they need it. By default, the value is set to True (so, not changing current behavior).
LM studio provide, by default, nomic-ai/nomic-embed-text-v1.5-GGUF for embedding. When calling the exposed API, with tiktoken_enabled=True, the client sends integers as embedding requests. The API expects Strings and it returns errors: [ERROR] 'input' field must be a string or an array of strings

baskaryan reviewed (commented):
what does typing this as Optional change? you can already set value to False (Optional gives you ability to set it to None)

karkaratz commented:
> what does typing this as Optional change? you can already set value to False (Optional gives you ability to set it to None)

You are right, it  has no point to make it optional.

Commit:
With this update, users can set as False tiktoken_enabled when they need it. By default, the value is set to True (so, not changing current behavior).
LM studio provide, by default, nomic-ai/nomic-embed-text-v1.5-GGUF for embedding. When calling the exposed API, with tiktoken_enabled=True, the client sends integers as embedding requests. The API expects Strings and it returns errors: [ERROR] 'input' field must be a string or an array of strings

baskaryan commented:
> > what does typing this as Optional change? you can already set value to False (Optional gives you ability to set it to None)
> 
> You are right, it has no point to make it optional. What you suggest to do? Close this PR and Open a New One?

what would new pr do? is a pr needed at all?

karkaratz commented:
> > > what does typing this as Optional change? you can already set value to False (Optional gives you ability to set it to None)
> > 
> > 
> > You are right, it has no point to make it optional. What you suggest to do? Close this PR and Open a New One?
> 
> what would new pr do? is a pr needed at all?

The conversation is too asynchronous. 😄 
I had modified the PR and I had removed the Optional.

hwchase17 commented:
im not really sure what this does, i think ` = True` is the same as ` = Field(default=True)`

if i'm wrong, feel free to reopen
